finesse/hardware/serial_manager.py

```diff
@@ -0,0 +1,99 @@
+"""Common functionality for opening and closing USB serial devices."""
+import logging
+from collections.abc import Callable
+
+from pubsub import pub
+from serial import Serial
+
+from ..config import DUMMY_DEVICE_PORT
+from .device_base import DeviceBase
+
+
+def make_device_factory(
+    device_factory: Callable[[Serial], DeviceBase],
+    dummy_device_factory: Callable[[], DeviceBase],
+) -> Callable[[str, int], DeviceBase]:
+    """Make a device factory function.
+
+    Args:
+        device_factory: A factory function taking a Serial object as an argument
+        dummy_device_factory: A factory function taking no arguments and returning a
+                              dummy device object
+    """
+
+    def create_device(port: str, baudrate: int) -> DeviceBase:
+        if port == DUMMY_DEVICE_PORT:
+            return dummy_device_factory()
+        else:
+            serial = Serial(port, baudrate)
+            return device_factory(serial)
+
+    return create_device
```

Comment:
This just moves the problem to a separate function, creating, on top of that, a nested function impossible to test directly. Also, for what I can tell, all those `device_factory` and `dummy_device_factory` often need `partial` functions with the arguments half populated. And, on top of that, the temperature controller, for example, has its own way of creating a serial connection (the `create` method), with the constructor accepting a `name` that is not used anywhere. 

I know we don't have time to spare, but I really think you need to go back to the drawing board and re-think the architecture of how serial devices are managed because, as it now, is really not clean. I'm happy to be involved in that discussion.